lib/block-supports/dimensions.php

```diff
@@ -21,10 +21,10 @@ function gutenberg_register_dimensions_support( $block_type ) {
 		return;
 	}
 
-	$has_spacing_support = gutenberg_block_has_support( $block_type, array( 'spacing' ), false );
-	// Future block supports such as height & width will be added here.
+	$has_dimensions_support = gutenberg_block_has_support( $block_type, array( '__experimentalDimensions' ), false );
```

Comment:
Sorry for being late for the conversation. I've noticed that in https://github.com/WordPress/gutenberg/pull/32392 we renamed a lot of things from spacing to dimensions. Couldn't look at it deeply yet, but wanted to flag that changing keys in `block.json` or `theme.json` is problematic if they already landed in WordPress core.

For example, here we're changing `spacing` to `__experimentalDimensions` in `block.json`. But `spacing` has already landed in WordPress core, so we need to keep `spacing` working as before. If we absolutely need to rename it, we can add some code that converts `spacing` into `dimensions`. If it's just a name preference, I'd advise to keep using what we have (spacing) and don't make our work harder for this.

At the UI level, it can be labeled differently if we need to and we also can change it as many times as we need.

FYI @jorgefilipecosta @youknowriad

Reply:
I've gone through a first pass at #32392 I may have found an issue https://github.com/WordPress/gutenberg/pull/32392/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.php#r686752790

If we keep the `spacing` key, I'd suggest we rename the things back to spacing internally, although at the UI level we name them "dimensions".

Reply:
> Couldn't look at it deeply yet, but wanted to flag that changing keys in block.json or theme.json is problematic if they already landed in WordPress core.

Thanks @nosolosw, I'd thought of that and thought I'd maintained that backwards compatibility. Happy to change whatever I've messed up. I'm missing where that is though.

> For example, here we're changing spacing to __experimentalDimensions in block.json. But spacing has already landed in WordPress core, so we need to keep spacing working as before.

I've actually not changed `spacing` to `__experimentalDimensions` in the block.json or theme.json. In the line below this comment you can still see the check for the `spacing` portion of the overall dimensions themed block support.

My plan was to introduce the new specific dimensions related supports under "dimensions" itself. It didn't make sense to me to have `height`, `width` etc under `spacing`.

Some points about where the separation of dimensions and spacing keys/support occurs are:
- `lib/class-wp-theme-json-gutenberg.php` adds `dimensions` in addition to, rather than replacing `spacing`
- `lib/theme.json` adds the setting for `dimensions.customHeight` separate to `spacing`
- `packages/block-editor/src/hooks/dimensions.js` checks separate block.json flags for `spacing` vs `__experimentalDimensions` support
- `packages/block-editor/src/hooks/style.js` dimensions key was added alongside spacing for the inline style generation
- The padding/margin styles are still under `spacing` within the block attributes style object
- The written tests also indicate that the spacing keys were still maintained.

Your [comment](https://github.com/WordPress/gutenberg/pull/32392#discussion_r686752790) regarding the key used when registering the block support in `dimensions.php` was well made and I have a fix for that specifically in https://github.com/WordPress/gutenberg/pull/34030.

Is there anything else I have missed? Thanks for your help as always!

Reply:
Thanks for the quick turnaround, Aaron. I have a [second thought](https://github.com/WordPress/gutenberg/pull/32392/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md&file-filters%5B%5D=.php&hide-deleted-files=true#r687501190) about the other PR. I can review this one properly once we clarify whether what I've brought up is an issue or not.